File: Tests/WebApiController/Models/OptionalPropertiesModel.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.Diagnostics.CodeAnalysis;

namespace WebApiController.Models
{
    public class OptionalPropertiesModel
    {
        [Required]
        public string RequiredString { get; set; }

        [Required, DefaultValue("")]
        public string RequiredNotNullableString { get; set; }

        [Required]
        public int RequiredInt { get; set; }

        public string OptionalString { get; set; }
        public int OptionalInt { get; set; }
    }
}

File: Tests/WebApiController/ClientApp/src/app/date/models/optional-properties-model.ts
/* eslint-disable */
// tslint:disable

export class OptionalPropertiesModel {
    public requiredString: string;
    public requiredNotNullableString: string = "";
    public requiredInt: number;
    public optionalString?: string;
    public optionalInt?: number;

    public constructor(init?: Partial<OptionalPropertiesModel>) {
        Object.assign(this, init);
    }
}

// outputid:627408ca-a818-4326-b843-415f5bbfb028
